app/addons/fauxton/navigation/container/NavBar.js

```diff
@@ -0,0 +1,77 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+import FauxtonAPI from "../../../../core/api";
+import React from "react";
+
+import ReactDOM from "react-dom";
+import Stores from "../stores";
+
+import NavBar from '../components/NavBar';
+
+const navBarStore = Stores.navBarStore;
+
+const NavBarContainer = React.createClass({
+
+  getStoreState () {
+    return {
+      navLinks: navBarStore.getNavLinks(),
+      bottomNavLinks: navBarStore.getBottomNavLinks(),
+      footerNavLinks: navBarStore.getFooterNavLinks(),
+      activeLink: navBarStore.getActiveLink(),
+      version: navBarStore.getVersion(),
+      isMinimized: navBarStore.isMinimized(),
+      isNavBarVisible: navBarStore.isNavBarVisible(),
+
+      isLoginSectionVisible: navBarStore.getIsLoginSectionVisible(),
+      isLoginVisibleInsteadOfLogout: navBarStore.getIsLoginVisibleInsteadOfLogout()
+    };
+  },
+
+  getInitialState () {
+    return this.getStoreState();
+  },
+
+  onChange () {
+    this.setState(this.getStoreState());
+  },
+
+  setMenuState () {
+    FauxtonAPI.Events.trigger(FauxtonAPI.constants.EVENTS.NAVBAR_SIZE_CHANGED, this.state.isMinimized);
```

Comment:
Is this trigger event still needed. Could we rather use a store instead. I would like to remove all our trigger events as much as possible.